plugin/metric/collector.go

```diff
@@ -0,0 +1,10 @@
+package metric
+
+import (
+	"context"
+)
+
+// Collector is the API message for metric collector.
+type Collector interface {
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#interfaces

Go interfaces generally belong in the package that uses values of the interface type, not the package that implements those values. ...

Reply:
shall we move `Collector` to /store and rename `StoreMetricCollector`?

Reply:
Yes, that's the intention. We do need a collector or counter struct in the store package for doing the COUNT and GROUP by stuff.